packages/react/src/feature-app-container.tsx

```diff
@@ -55,14 +55,23 @@ export interface FeatureAppContainerProps {
 type InternalFeatureAppContainerProps = FeatureAppContainerProps &
   FeatureHubContextValue;
 
+interface InternalFeatureAppContainerState {
+  hasError: boolean;
+}
+
 const inBrowser =
   typeof window === 'object' &&
   typeof document === 'object' &&
   document.nodeType === 9;
 
 class InternalFeatureAppContainer extends React.PureComponent<
-  InternalFeatureAppContainerProps
+  InternalFeatureAppContainerProps,
+  InternalFeatureAppContainerState
 > {
+  public readonly state: InternalFeatureAppContainerState = {
+    hasError: false
```

Comment:
Maybe rename to `hasLifecycleError` because rendering errors are handled directly in the render method.

Reply:
Maybe we should remember the complete lifecycle error instead of a boolean to re-throw it below.

Reply:
See https://github.com/sinnerschrader/feature-hub/pull/352#discussion_r255296839